modules/wallet/persist_legacy.go

```diff
@@ -1,177 +0,0 @@
-	for _, skey := range savedKeys {
-		// Skip this key if it's already known to the wallet.
-		_, exists := w.keys[skey.UnlockConditions.UnlockHash()]
-		if exists {
-			continue
-		}
-
-		// Create an entry in w.keys for each savedKey.
-		w.keys[skey.UnlockConditions.UnlockHash()] = &key{
-			spendable:        height >= skey.UnlockConditions.Timelock,
-			unlockConditions: skey.UnlockConditions,
-			secretKey:        skey.SecretKey,
-			outputs:          make(map[types.SiacoinOutputID]*knownOutput),
-		}
-
-		// If Timelock != 0, also add to set of timelockedKeys.
-		if tl := skey.UnlockConditions.Timelock; tl != 0 {
-			w.timelockedKeys[tl] = append(w.timelockedKeys[tl], skey.UnlockConditions.UnlockHash())
-		}
-
-		if skey.Visible {
-			w.visibleAddresses[skey.UnlockConditions.UnlockHash()] = struct{}{}
-		}
-	}
-
-	// If there are no visible addresses, create one.
-	if len(w.visibleAddresses) == 0 {
-		_, _, err := w.coinAddress(true)
-		if err != nil {
-			return err
-		}
-	}
-
-	return nil
-}
-
-// loadWallet pulls a wallet from disk into memory, merging it with whatever
-// wallet is already in memory. The result is a combined wallet that has all of
-// the addresses.
-func (w *Wallet) loadWallet(filepath string) error {
-	var savedKeys []savedKey
-	err := encoding.ReadFile(filepath, &savedKeys)
-	if err != nil {
-		return err
-	}
-	err = w.loadKeys(savedKeys)
-	if err != nil {
-		return err
-	}
-	return nil
-}
-
-// loadSiafundTracking loads siafund addresses for tracking.
-func (w *Wallet) loadSiafundTracking(filepath string) error {
-	// Load the siafunds file, which is intentionally called 'outputs.dat'.
-	var siafundAddresses []types.UnlockHash
-	err := encoding.ReadFile(filepath, &siafundAddresses)
-	if err != nil {
-		return err
-	}
-
-	// Load the addresses into the wallet.
-	for _, sa := range siafundAddresses {
-		w.siafundAddresses[sa] = struct{}{}
-	}
-	return nil
-}
-
-// load reads the contents of a wallet from a file.
-func (w *Wallet) load() error {
-	err := w.loadWallet(filepath.Join(w.persistDir, "wallet.dat"))
-	if err != nil {
-		// try loading the backup
-		// TODO: display/log a warning
-		err = w.loadWallet(filepath.Join(w.persistDir, "wallet.backup"))
-		if err != nil {
-			return err
-		}
-	}
-
-	// Load the siafunds file, which is intentionally called 'outputs.dat'.
-	err = w.loadSiafundTracking(filepath.Join(w.persistDir, "outputs.dat"))
-	if err != nil {
-		// try loading the backup
-		// TODO: display/log a warning?
-		err = w.loadSiafundTracking(filepath.Join(w.persistDir, "outputs.backup"))
-		if err != nil {
-			return err
-		}
-	}
-
-	return w.save()
-}
-
-// MergeWallet merges another wallet with the already-loaded wallet, creating a
-// new wallet that contains all of the addresses from each. This is useful for
-// loading backups.
-func (w *Wallet) MergeWallet(filepath string) error {
-	err := w.loadWallet(filepath)
-	if err != nil {
-		return err
-	}
-	return w.save()
-}
-*/
```

Comment:
well, I'm actually referring to a different type of compatibility code.

The `compatibility` package is for upgrading persist files between `siad` versions, so everything in that package should approximately follow the method of `compatV1.0.2ToV1.0.3(persistFolder)`.

I don't think it makes as much sense to have the wallet loading code in the compatibility package, because the method of loading old wallets is going to change if the internals of the wallet change. It's not something you can easily chain because you aren't changing the 033x file itself, you're just using the contents to learn some unlock hashes which you are supposed to be tracking.